src/bennu/devices/modules/comms/iec60870-5/module/Server.cpp

```diff
@@ -315,6 +526,41 @@ bool Server::asduHandler(void *parameter, IMasterConnection connection, CS101_AS
 
         return true;
     }
+    else if (CS101_ASDU_getTypeID(asdu) == C_DC_NA_1)
+    {
+        printf("received double command\n");
+
+        if (CS101_ASDU_getCOT(asdu) == CS101_COT_ACTIVATION)
+        {
+            InformationObject io = CS101_ASDU_getElement(asdu, 0);
+
+            if (io)
+            {
+                // Send activation confirmation (application layer ACK)
+                IMasterConnection_sendACT_CON(connection, asdu, false);
+
+                DoubleCommand dc = (DoubleCommand)io;
+                uint16_t addr = InformationObject_getObjectAddress(io);
+                int state = DoubleCommand_getState(dc);
+                printf("IOA: %i switch to %i\n", addr, state);
+                gServer->writeBinary(addr, state);
+                // Send activation termination
+                CS101_ASDU_setCOT(asdu, CS101_COT_ACTIVATION_TERMINATION);
+                InformationObject_destroy(io);
+            }
+            else
+            {
+                printf("ERROR: message has no valid information object\n");
+                return true;
```

Comment:
If this is an error, should it be returning true?

Reply:
I don't know that anything checks this return but fixed anyway